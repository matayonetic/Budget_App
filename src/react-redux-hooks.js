import {
  useSelector as originalUseSelector,
  useDispatch as originalUseDispatch,
} from "react-redux";

export const useDispatch = () => originalUseDispatch();
export const useSelector = (state) => originalUseSelector(state);
