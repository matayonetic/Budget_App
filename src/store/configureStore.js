import { legacy_createStore as createStore, combineReducers } from "redux";

// Reducers
import expensesReducer from "../reducers/expensesReducer";
import filtersReducer from "../reducers/filtersReducer";

// Store
const configureStore = () => {
  const store = createStore(
    combineReducers({
      expenses: expensesReducer,
      filters: filtersReducer,
    }),
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  );
  return store;
};

export { configureStore as default };
