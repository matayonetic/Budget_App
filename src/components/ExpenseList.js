import React from "react";
import { useSelector } from "../react-redux-hooks";
import ExpenseListItems from "./ExpenseListItems";
import getVisibleExpenses from "../selectors/getVisibleExpenses";

const ExpenseList = () => {
  const expenses = useSelector((state) =>
    getVisibleExpenses(state.expenses, state.filters)
  );
  return (
    <div>
      <h2>Expense List</h2>
      {expenses.length > 0 ? (
        expenses.map((expense) => {
          return <ExpenseListItems key={expense.id} {...expense} />;
        })
      ) : (
        <p>No data found</p>
      )}
    </div>
  );
};

export default ExpenseList;
