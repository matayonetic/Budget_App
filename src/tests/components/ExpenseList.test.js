import React from "react";
import { shallow } from "enzyme";

import expenses from "../fixtures/expenses";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import * as ReactReduxHooks from "../../react-redux-hooks";

import ExpenseList from "../../components/ExpenseList";

// Render Expense List
test("Should render expense list", () => {
  //
  let store = configureStore([thunk])(expenses);
  jest
    .spyOn(ReactReduxHooks, "useSelector")
    .mockImplementation(() => store.getState());
    
  const wrapper = shallow(<ExpenseList />);
  expect(wrapper).toMatchSnapshot();
});

// Affirm Empty Expense List
test("Should render an empty expense list", () => {
  let store = configureStore([thunk])([]);
  jest
    .spyOn(ReactReduxHooks, "useSelector")
    .mockImplementation(() => store.getState());
  const wrapper = shallow(<ExpenseList />);
  expect(wrapper.find("p").text()).toBe("No data found");
  expect(wrapper).toMatchSnapshot();
});
