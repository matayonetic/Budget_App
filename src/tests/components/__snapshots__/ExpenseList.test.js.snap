// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render an empty expense list 1`] = `
<div>
  <h2>
    Expense List
  </h2>
  <p>
    No data found
  </p>
</div>
`;

exports[`Should render expense list 1`] = `
<div>
  <h2>
    Expense List
  </h2>
  <Connect(ExpenseListItems)
    amount={1}
    createdAt={-86400000}
    description="Food"
    id="abc1"
    key="abc1"
    note="a"
  />
  <Connect(ExpenseListItems)
    amount={2}
    createdAt={0}
    description="Drinks"
    id="abc2"
    key="abc2"
    note="b"
  />
  <Connect(ExpenseListItems)
    amount={3}
    createdAt={172800000}
    description="Party"
    id="abc3"
    key="abc3"
    note="c"
  />
</div>
`;
